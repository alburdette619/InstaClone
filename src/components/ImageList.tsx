import React, { Component } from 'react';
import { FlatList, StyleSheet, View } from 'react-native';
import { ImageListProps } from '../types/interfaces';
import LightboxImage from './LightboxImage';

interface ImageListState {
  imageSize: number;
}

class ImageList extends Component<ImageListProps, ImageListState> {
  constructor(props: ImageListProps) {
    super(props);

    this.state = {
      imageSize: 0,
    };
  }

  render() {
    const { sources, navigation } = this.props;
    const { imageSize } = this.state;

    return (
      <View
        style={styles.container}
        onLayout={({
          nativeEvent: {
            layout: { width },
          },
        }) => {
          this.setState({ imageSize: width / 3 });
        }}
      >
        <FlatList
          data={sources}
          renderItem={({ item }) => (
            <View
              style={{
                width: imageSize,
                height: imageSize,
              }}
            >
              <LightboxImage source={item} navigation={navigation} />
            </View>
          )}
          numColumns={3}
          keyExtractor={(item, index) => `${item.uri}${index}`}
          extraData={this.state.imageSize}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: { flex: 1 },
});

export default ImageList;
