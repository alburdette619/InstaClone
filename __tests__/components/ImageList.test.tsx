import * as React from 'react';
import { ImageURISource } from 'react-native';
import * as renderer from 'react-test-renderer';
import ImageList from '../../src/components/ImageList';
import * as mocks from '../__mocks__/mocks';

it('renders without crashing', () => {
  const sources: [ImageURISource, ImageURISource, ImageURISource] = [
    mocks.source,
    mocks.source,
    mocks.source,
  ];

  const rendered = renderer
    .create(<ImageList sources={sources} navigation={mocks.nav} />)
    .toJSON();
  expect(rendered).toMatchSnapshot();
});
