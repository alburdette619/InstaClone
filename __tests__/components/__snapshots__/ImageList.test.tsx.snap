// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "uri": "https://picsum.photos/200/300/?random",
        },
        Object {
          "uri": "https://picsum.photos/200/300/?random",
        },
        Object {
          "uri": "https://picsum.photos/200/300/?random",
        },
      ]
    }
    disableVirtualization={false}
    extraData={0}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={3}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "height": 0,
                "width": 0,
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "borderColor": "white",
                  "borderWidth": 1,
                  "flex": 1,
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": "https://picsum.photos/200/300/?random",
                  }
                }
                style={
                  Object {
                    "flex": 1,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "height": 0,
                "width": 0,
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "borderColor": "white",
                  "borderWidth": 1,
                  "flex": 1,
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": "https://picsum.photos/200/300/?random",
                  }
                }
                style={
                  Object {
                    "flex": 1,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "height": 0,
                "width": 0,
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "borderColor": "white",
                  "borderWidth": 1,
                  "flex": 1,
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": "https://picsum.photos/200/300/?random",
                  }
                }
                style={
                  Object {
                    "flex": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
